rundeckapp/grails-app/services/rundeck/services/asyncimport/AsyncImportStatusDTO.groovy

```diff
@@ -0,0 +1,66 @@
+package rundeck.services.asyncimport
+
+import groovy.transform.builder.Builder
+import org.apache.commons.beanutils.PropertyUtils
+
+@Builder
+class AsyncImportStatusDTO {
+    String tempFilepath;
+    String projectName;
+    Integer milestoneNumber;
+    String milestone;
+    String lastUpdated;
+    String errors;
+    String lastUpdate;
+    String jobUuidOption;
+
+    AsyncImportStatusDTO(){}
+
+    AsyncImportStatusDTO(
+            String projectName,
+            int milestoneNumber
+    ){
+        this.projectName = projectName
+        this.milestoneNumber = milestoneNumber
+    }
+
+    AsyncImportStatusDTO(
+            AsyncImportStatusDTO newStatus
+    ){
+        this.tempFilepath = newStatus.tempFilepath
+        this.projectName = newStatus.projectName
+        this.milestoneNumber = newStatus.milestoneNumber
+        this.milestone = newStatus.milestone
+        this.lastUpdated = newStatus.lastUpdated
+        this.errors = newStatus.errors
+        this.lastUpdate = newStatus.lastUpdate
+        this.jobUuidOption = newStatus.jobUuidOption
+    }
+
+    /**
+     * Replaces all null properties in destinationDTO with the properties in sourceDTO that match with null keys of destinationDTO,
+     * be sure that destinationDTO has null the values that you DONT want to change of status file in DB.
+     *
+     * @param destinationDTO
+     * @param sourceDTO
+     * @return
+     */
+    static def replacePropsInTargetDTO(AsyncImportStatusDTO destinationDTO, AsyncImportStatusDTO sourceDTO) {
```

Comment:
why is this needed? theres already a constructor to assign values from an existing AsyncImportStatusDTO. Try catch is not doing anything and service calling this method would not be aware of a failure

Reply:
Because of the design of the Updater, we want to combine the status in DB with a new status constructed in runtime; if we pass a new status, the null values in the new one will override the existing with a null. Maybe if i change the method signature to "combineStatus" or so, would be helpful.

Reply:
Replace logic with if's